src/NHSD.BuyingCatalogue.Organisations.Api/Repositories/OdsRepository.cs

```diff
@@ -48,8 +46,12 @@ private static bool IsActive(OdsResponseOrganisation organisation)
 
         private bool IsBuyerOrganisation(OdsResponseOrganisation organisation)
         {
-            return organisation.Roles.Role.Any(role =>
-                 role.id.Equals(_buyerOrganisationRoleId, StringComparison.OrdinalIgnoreCase));
+            return _settings.BuyerOrganisationRoleIds.Contains(GetPrimaryRoleId(organisation));
+        }
+
+        private static string GetPrimaryRoleId(OdsResponseOrganisation organisation)
+        {
+            return organisation.Roles.Role.Where(r => r.primaryRole).Select(r => r.id).FirstOrDefault();
```

Comment:
This is fine. However, are you aware it can also be written as below?

```c#
return organisation.Roles.Role.FirstOrDefault(r => r.primaryRole)?.id;
```
I'd probably go with the version here just because it's slightly simpler. However, it's entirely up to you which version you go with – they both achieve the same result.

Reply:
I was not aware, thanks for pointing it out! I much prefer your version.